core/src/main/java/org/apache/accumulo/core/clientImpl/NamespaceOperationsImpl.java

```diff
@@ -227,6 +216,34 @@ public void modifyProperties(final String namespace,
     }
   }
 
+  public void modifyProperties(final String namespace,
+      final Consumer<Map<String,String>> mapMutator)
+      throws AccumuloException, AccumuloSecurityException, NamespaceNotFoundException {
+    EXISTING_NAMESPACE_NAME.validate(namespace);
+    checkArgument(mapMutator != null, "mapMutator is null");
+
+    Retry retry =
+        Retry.builder().infiniteRetries().retryAfter(25, MILLISECONDS).incrementBy(25, MILLISECONDS)
+            .maxWait(30, SECONDS).backOffFactor(1.5).logInterval(3, MINUTES).createRetry();
+
+    while (true) {
+      try {
+        tryToModifyProperties(namespace, mapMutator);
+        break;
+      } catch (ConcurrentModificationException cme) {
+        try {
+          retry.logRetry(log, "Unable to modify namespace properties for " + namespace
```

Comment:
Maybe it could read - "modification attempt failed because of a concurrent modification - will retry"

Reply:
Agree that we should add the fact the fact that we are going to retry here and in other places where Retry.logRetry is called.

Reply:
I made a top level comment about this.

Reply:
I think this can be resolved based on the top level comment.